src/CachePlugin.php

```diff
@@ -273,12 +276,17 @@ private function configureOptions(OptionsResolver $resolver)
             'default_ttl' => 0,
             'respect_cache_headers' => true,
             'hash_algo' => 'sha1',
+            'methods' => ['GET', 'HEAD'],
         ]);
 
         $resolver->setAllowedTypes('cache_lifetime', ['int', 'null']);
         $resolver->setAllowedTypes('default_ttl', ['int', 'null']);
         $resolver->setAllowedTypes('respect_cache_headers', 'bool');
+        $resolver->setAllowedTypes('methods', 'array');
         $resolver->setAllowedValues('hash_algo', hash_algos());
+        $resolver->setAllowedValues('methods', function ($value) {
+            return 0 === count(array_diff($value, ['GET', 'HEAD', 'POST']));
```

Comment:
I agree. Question is since RFC2616 says valid request method can be pretty much any string (not sure if A..Z or alphanumeric) should any request method be allowed or just the common methods defined in [RFC2616-sec9](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html)